pages/admin/dashboard.js

```diff
@@ -0,0 +1,229 @@
+                    gameArrayState[gamePin].state === "GAME_STATE-WAITING" &&
+                      stateToggle(gamePin, "GAME_QUESTION_1");
+                  }}
+                  color="primary"
+                  id={styles.enterButton}
+                >
+                  {gameArrayState[gamePin].state === "GAME_STATE-WAITING"
+                    ? "Start Game"
+                    : "View Game"}
+                </Button>
+              </Link>
+
+              <Link href="dashboard">
+                <Button
+                  variant="contained"
+                  onClick={() => stateToggle(gamePin, "END")}
+                  color="primary"
+                  id={styles.enterButton}
+                >
+                  End game
+                </Button>
+              </Link>
+            </div>
+
+            <Typography variant="h5" style={{ marginTop: "20px" }} gutterBottom>
+              Game state:{" "}
+              {gameArrayState[gamePin].state.replace("GAME_STATE-", "")}
+            </Typography>
+
+            <Card>
+              <CardContent id={styles.nameContainer}>
+                {gameArrayState[gamePin].users !== undefined &&
+                  Object.keys(gameArrayState[gamePin].users).map(function (
+                    key
+                  ) {
+                    return (
+                      <div style={{ display: "flex", flexDirection: "row" }}>
+                        <div
+                          onClick={() => shadowToggle(gamePin, key)}
+                          style={{
+                            textDecoration:
+                              gameArrayState[gamePin].users[key].sban !==
+                                undefined &&
+                              gameArrayState[gamePin].users[key].sban
+                                ? "line-through"
+                                : "",
+                          }}
+                          id={styles.username}
+                        >
+                          {gameArrayState[gamePin].users[key].name}
+                        </div>
+                      </div>
+                    );
+                  })}
+              </CardContent>
+            </Card>
+          </Container>
+        </div>
+      ) : (
+        login && (
+          <div>
+            <h1>
+              Hey, {firebase.auth().currentUser.displayName}! Welcome to the
+              Admin Dashboard
+            </h1>
+            <h1 onClick={() => signOutUser()}>Sign out</h1>
+            <Container id={styles.listContainer} style={{ minHeight: "60vh" }}>
+              <Typography variant="h2" id={styles.title} gutterBottom>
+                View all games
+              </Typography>
+              <Card className={styles.gamePinCard}>
+                <CardContent className={styles.pinContainer}>
+                  {listOfGames.map((el) => {
+                    return (
+                      <div style={{ display: "flex", flexDirection: "row" }}>
```

Comment:
need to add `key` property here

https://adhithiravi.medium.com/why-do-i-need-keys-in-react-lists-dbb522188bbb